games/static/games/proto.js

```diff
@@ -62,14 +86,15 @@ class Square {
         this.div.classList.add(cssClass)
     }
     setEmpty() {
-        this.div.className= ""
+        this.div.className = ""
         this.character = null
     }
 
     setGround() {
         this.div.classList.add('ground')
     }
     hasAlien() {
+        if (this.character != null && this.character.health === 0) this.setEmpty()
```

Comment:
styling - i prefer new line after the if (makes it easier to read)
and the name is misleading.
from reading the name i dont expect anything to change on the object
i expect it to check if this square has alient or not
but it does something on the way and returns an answer.
feels like setEmpty should be called from somewhere else
maybe when rocket hits the alien and health go down to 0
but not here

Reply:
this function really made everything work perfectly so i left hasAlien() without the action just return yes or no and i made a new method on square checkForAlienAndRemoveIfDead()

Reply:
ok lets think it over
why does it need to be in one function ?
you can have a function removeDeadAlien and the caller can then use hasAlien

Reply:
you are so right!!!!!!! i did it